docs/security/hardening-guide.md

```diff
@@ -464,6 +560,165 @@ sudo systemctl daemon-reload
 sudo systemctl restart k3s.service
 ```
 
+## Configuration for Kubernetes Components
+
+
+The configuration below should be placed in the [configuration file](../installation/configuration.md#configuration-file), and contains all the necessary remediations to harden the Kubernetes components.
+
+
+<Tabs>
+<TabItem value="v1.25 and Newer" default>
+
+```yaml
+protect-kernel-defaults: true
+secrets-encryption: true
+kube-apiserver-arg:
+  - 'enable-admission-plugins=NodeRestriction,NamespaceLifecycle,ServiceAccount'
+  - 'admission-control-config-file=/var/lib/rancher/k3s/server/psa.yaml'
+  - 'audit-log-path=/var/lib/rancher/k3s/server/logs/audit.log'
+  - 'audit-policy-file=/var/lib/rancher/k3s/server/audit.yaml'
+  - 'audit-log-maxage=30'
+  - 'audit-log-maxbackup=10'
+  - 'audit-log-maxsize=100'
+  - 'request-timeout=300s'
+  - 'service-account-lookup=true'
+kube-controller-manager-arg:
+  - 'terminated-pod-gc-threshold=10'
+  - 'use-service-account-credentials=true'
+kubelet-arg:
+  - 'streaming-connection-idle-timeout=5m'
+  - 'make-iptables-util-chains=true'
+```
+
+</TabItem>
+
+<TabItem value="v1.24 and Older" default>
+
+```yaml
+protect-kernel-defaults: true
+secrets-encryption: true
+kube-apiserver-arg:
+  - 'enable-admission-plugins=NodeRestriction,PodSecurityPolicy,NamespaceLifecycle,ServiceAccount'
```

Comment:
same question

Reply:
This line cannot be removed from the 1.24 and Older, as older version of Kubernetes such as v1.22 did not enable NodeRestriciton or PodSecurityPolicy by default.